fms_fsdp/utils/dataloader_utils.py

```diff
@@ -19,6 +19,18 @@
 }
 
 
+def causal_lm(data_seq, prompt_len=1):
+    """
+    Perform causal language modeling by right-shifting the input sequence.
+    Sets first prompt_len tokens to be ignored by the loss.
+    """
+    data_seq = torch.IntTensor(data_seq)
```

Comment:
do we need to add device to this?

Reply:
nit: I think it's generally considered preferable "pytorch style" to use torch.tensor(..., dtype=torch.int) over IntTensor

Reply:
Will do!

Reply:
This doesn't need device - it's internal to the `torch.utils.data.Dataset` pipeline, and all pipeline stuff is on cpu until the data leaves the pipeline (that's a general pytorch rule, not just us)